tslearn/metrics/dtw_variants.py

```diff
@@ -1365,3 +1365,506 @@ def lb_envelope(ts, radius=1):
+
+    Parameters
+    ----------
+    s1 : array, shape = (sz1,)
+        First time series.
+
+    s2 : array, shape = (sz2,)
+        Second time series.
+
+    eps : float
+        Matching threshold.
+
+    mask : array, shape = (sz1, sz2)
+        Mask. Unconsidered cells must have infinite values.
+
+    Returns
+    -------
+    lcss_score : float
+        Longest Common Subsequence similarity score between both time series.
+
+    """
+    l1 = s1.shape[0]
+    l2 = s2.shape[0]
+    acc_cost_mat = numpy.full((l1 + 1, l2 + 1), 0)
+
+    for i in range(1, l1 + 1):
+        for j in range(1, l2 + 1):
+            if numpy.isfinite(mask[i - 1, j - 1]):
+                if numpy.sqrt(_local_squared_dist(s1[i - 1], s2[j - 1])) <= eps:
+                    acc_cost_mat[i][j] = 1 + acc_cost_mat[i - 1][j - 1]
+                else:
+                    acc_cost_mat[i][j] = max(acc_cost_mat[i][j - 1],
+                                             acc_cost_mat[i - 1][j])
+
+    return acc_cost_mat
+
+
+@njit(nogil=True)
+def njit_lcss(s1, s2, eps, mask):
+    """Compute the longest common subsequence score between two time series.
+
+    Parameters
+    ----------
+    s1 : array, shape = (sz1,)
+        First time series.
+
+    s2 : array, shape = (sz2,)
+        Second time series
+
+    eps : float (default: 1.)
+        Maximum matching distance threshold.
+
+    mask : array, shape = (sz1, sz2)
+        Mask. Unconsidered cells must have infinite values.
+
+    Returns
+    -------
+    lcss_score : float
+        Longest Common Subsquence score between both time series.
+    """
+    l1 = s1.shape[0]
+    l2 = s2.shape[0]
+    acc_cost_mat = njit_lcss_accumulated_matrix(s1, s2, eps, mask)
+
+    return float(acc_cost_mat[-1][-1]) / min([l1, l2])
+
+
+def lcss(s1, s2, eps=1., global_constraint=None, sakoe_chiba_radius=None,
+        itakura_max_slope=None):
+    r"""Compute the Longest Common Subsequence (LCSS) similarity measure
+    between (possibly multidimensional) time series and return both the
+    path and the similarity.
+
+    LCSS is computed by matching indexes that are met up until the eps
+    threshold, so it leaves some points unmatched and focuses on the
+    similar parts of two sequences. The matching can occur even if the
+    time indexes are different, regulated through the delta parameter
+    that defines how far it can go. To retrieve a meaningful similarity
+    value from the length of the longest common subsequence, the
+    percentage of that value regarding the length of the shortest time
+    series is returned.
+
+    According to this definition, the values returned by LCSS range from
+    0 to 1, the highest value taken when two time series fully match,
+    and vice-versa. It is not required that both time series share the
+    same size, but they must be the same dimension. LCSS was originally
+    presented in [1]_ and is discussed in more details in our
+    :ref:`dedicated user-guide page <lcss>`.
```

Comment:
there is no such page

Reply:
This page is indeed missing. I will be adding it today.